mw/notification.d.ts

```diff
@@ -22,11 +22,7 @@ declare global {
         function notify(
             message: string | JQuery | HTMLElement | HTMLElement[],
             options?: Partial<typeof notification.defaults>
-        ): {
-            pause: () => void;
-            resume: () => void;
-            close: () => void;
-        };
+        ): ReturnType<typeof mw.loader.using>;
```

Comment:
I think you mean JQuery.Promise<Notification>

Reply:
well, it returns the return value of mw.loader.using. mw.loader.using itself actually returns a jquery promise object - which is why we're able to call `.then` on the result (notice also the JSDoc says it returns jQuery.Promise). In typescript, this is represented as `JQuery.Promise`. 

I do stand corrected though that it's not a promise wrapped on a notification object. A callback is used in the code instead of `.then()` chaining, so the promise ends up wrapping an internal function which seems unintentional and not worthy of being typed. So we should type it as `JQuery.Promise<void>`.